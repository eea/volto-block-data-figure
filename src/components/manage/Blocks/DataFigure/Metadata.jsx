/**
 * Metadata View.
 * @module components/manage/Blocks/DataFigure/Metadata
 */
import React from 'react';
import PropTypes from 'prop-types';
import {
  Header,
  Icon,
  Image,
  Segment,
  Menu,
  Sidebar,
  Container,
} from 'semantic-ui-react';

/**
 * Metadata  class.
 * @class Metadata
 * @extends Component
 */
const Metadata = ({ visible, data }) => {
  return (
    <Sidebar
      as={Menu}
      animation="overlay right"
      onHide={() => {}}
      vertical
      visible={visible}
      width="very wide"
    >
      <Segment basic>
        <div
          dangerouslySetInnerHTML={{
            __html: data.metadata?.dataSources,
          }}
        />
        <Header>Geographic coverage</Header>
        <div
          dangerouslySetInnerHTML={{
            __html: data.metadata?.geoCoverage,
          }}
        />

        <Header>Temporal coverage</Header>
        <div
          dangerouslySetInnerHTML={{
            __html: data.temporal.label,
          }}
        />
      </Segment>
    </Sidebar>
  );
};
/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
Metadata.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default Metadata;
