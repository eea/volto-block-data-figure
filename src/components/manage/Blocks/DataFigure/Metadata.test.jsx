import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-intl-redux';
import configureStore from 'redux-mock-store';
import Metadata from './Metadata';
import voltoPackageJson from '@plone/volto/../package.json';
import '@testing-library/jest-dom/extend-expect';

jest.mock('@eeacms/volto-widget-temporal-coverage/components', () => ({
  TemporalWidgetView: () => <div>TemporalWidgetView</div>,
}));

const mockStore = configureStore([]);

const store = mockStore({
  screen: {
    page: {
      width: 768,
    },
  },
  intl: {
    locale: 'en',
    messages: {},
    formatMessage: jest.fn(),
  },
});

const voltoVersion = voltoPackageJson.version;
const volto17 = voltoVersion.startsWith('17');

describe('<Metadata />', () => {
  it(`renders without crashing ${volto17 ? 'Volto17' : 'Volto16'}`, () => {
    const data = {
      data_provenance: {
        data: [
          {
            '@id': '1',
            link: 'https://www.example.com',
            title: 'Data source',
            organization: 'Example Organization',
          },
        ],
      },
      geolocation: [],
      temporal: [],
    };
    const { container } = render(
      <Provider store={store}>
        <Metadata data={data} visible={true} />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it(`renders data sources correctly ${
    volto17 ? 'Volto17' : 'Volto16'
  }`, () => {
    const data = {
      data_provenance: {
        data: [
          {
            '@id': '1',
            link: 'https://www.example.com/1',
            title: 'Data source 1',
            organization: 'Example Organization',
          },
          {
            '@id': '2',
            link: 'https://www.example.com/2',
            title: 'Data source 2',
            organization: 'Example Organization',
          },
        ],
      },
      geolocation: [{ label: 'Location 1' }, { label: 'Location 2' }],
      temporal: ['2023-01-01', '2023-12-31'],
    };
    const { container } = render(
      <Provider store={store}>
        <Metadata data={data} visible={true} />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders without crashing without metadata', () => {
    const data = {
      metadata: undefined,
      geolocation: [],
      temporal: [],
    };
    const { container } = render(
      <Provider store={store}>
        <Metadata data={data} visible={true} />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders without crashing without data_provenance', () => {
    const data = {
      data_provenance: undefined,
      geolocation: [],
      temporal: [],
    };
    const { container } = render(
      <Provider store={store}>
        <Metadata data={data} visible={true} />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders without crashing without links', () => {
    const data = {
      data_provenance: {
        data: [
          {
            '@id': '1',
            title: 'Test Data Sources',
            link: undefined,
          },
        ],
      },
      geolocation: undefined,
      temporal: [],
    };
    const { container } = render(
      <Provider store={store}>
        <Metadata data={data} visible={true} />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
