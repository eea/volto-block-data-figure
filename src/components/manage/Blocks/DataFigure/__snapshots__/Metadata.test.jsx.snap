// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Metadata /> renders data sources correctly Volto16 1`] = `
<div>
  <div
    class="ui vertical ui overlay right very wide visible sidebar metadata-sidebar menu"
  >
    <div
      class="ui raised segments"
    >
      <div
        class="ui segment"
      >
        <h2
          class="ui header data-figure-block-header"
        >
          Metadata
        </h2>
      </div>
      <div
        class="ui segment"
      >
        <h3
          class="ui header data-figure-block-header"
        >
          Data Sources:
        </h3>
        <ul
          class="ui bulleted list widget"
          role="list"
        >
          <li
            class="horizontal"
            role="listitem"
          >
            <div>
              <a
                href="https://www.example.com/1"
                rel="noopener noreferrer"
                target="_blank"
              >
                Data source 1
              </a>
            </div>
          </li>
          <li
            class="horizontal"
            role="listitem"
          >
            <div>
              <a
                href="https://www.example.com/2"
                rel="noopener noreferrer"
                target="_blank"
              >
                Data source 2
              </a>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="ui segment"
      >
        <h3
          class="ui header data-figure-block-header"
        >
          Geographic coverage:
        </h3>
        <ul>
          <div
            class="ui two column grid"
          >
            <div
              class="row"
            >
              <div
                class="column"
              >
                <li>
                  Location 1
                </li>
              </div>
              <div
                class="column"
              >
                <li>
                  Location 2
                </li>
              </div>
            </div>
          </div>
        </ul>
      </div>
      <div
        class="ui segment"
      >
        <h3
          class="ui header data-figure-block-header"
        >
          Temporal coverage:
        </h3>
        <div>
          TemporalWidgetView
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Metadata /> renders data sources correctly Volto17 1`] = `
<div>
  <div
    class="ui vertical ui overlay right very wide visible sidebar metadata-sidebar menu"
  >
    <div
      class="ui raised segments"
    >
      <div
        class="ui segment"
      >
        <h2
          class="ui header data-figure-block-header"
        >
          Metadata
        </h2>
      </div>
      <div
        class="ui segment"
      >
        <h3
          class="ui header data-figure-block-header"
        >
          Data Sources:
        </h3>
        <ul
          class="ui bulleted list widget"
          role="list"
        >
          <li
            class="horizontal"
            role="listitem"
          >
            <div>
              <a
                class="external"
                href="https://www.example.com/1"
                rel="noopener noreferrer"
                target="_blank"
              >
                Data source 1
              </a>
            </div>
          </li>
          <li
            class="horizontal"
            role="listitem"
          >
            <div>
              <a
                class="external"
                href="https://www.example.com/2"
                rel="noopener noreferrer"
                target="_blank"
              >
                Data source 2
              </a>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="ui segment"
      >
        <h3
          class="ui header data-figure-block-header"
        >
          Geographic coverage:
        </h3>
        <ul>
          <div
            class="ui two column grid"
          >
            <div
              class="row"
            >
              <div
                class="column"
              >
                <li>
                  Location 1
                </li>
              </div>
              <div
                class="column"
              >
                <li>
                  Location 2
                </li>
              </div>
            </div>
          </div>
        </ul>
      </div>
      <div
        class="ui segment"
      >
        <h3
          class="ui header data-figure-block-header"
        >
          Temporal coverage:
        </h3>
        <div>
          TemporalWidgetView
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Metadata /> renders without crashing Volto16 1`] = `
<div>
  <div
    class="ui vertical ui overlay right very wide visible sidebar metadata-sidebar menu"
  >
    <div
      class="ui raised segments"
    >
      <div
        class="ui segment"
      >
        <h2
          class="ui header data-figure-block-header"
        >
          Metadata
        </h2>
      </div>
      <div
        class="ui segment"
      >
        <h3
          class="ui header data-figure-block-header"
        >
          Data Sources:
        </h3>
        <ul
          class="ui bulleted list widget"
          role="list"
        >
          <li
            class="horizontal"
            role="listitem"
          >
            <div>
              <a
                href="https://www.example.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                Data source
              </a>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Metadata /> renders without crashing Volto17 1`] = `
<div>
  <div
    class="ui vertical ui overlay right very wide visible sidebar metadata-sidebar menu"
  >
    <div
      class="ui raised segments"
    >
      <div
        class="ui segment"
      >
        <h2
          class="ui header data-figure-block-header"
        >
          Metadata
        </h2>
      </div>
      <div
        class="ui segment"
      >
        <h3
          class="ui header data-figure-block-header"
        >
          Data Sources:
        </h3>
        <ul
          class="ui bulleted list widget"
          role="list"
        >
          <li
            class="horizontal"
            role="listitem"
          >
            <div>
              <a
                class="external"
                href="https://www.example.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                Data source
              </a>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Metadata /> renders without crashing without data_provenance 1`] = `
<div>
  <div
    class="ui vertical ui overlay right very wide visible sidebar metadata-sidebar menu"
  >
    <div
      class="ui raised segments"
    >
      <div
        class="ui segment"
      >
        <h2
          class="ui header data-figure-block-header"
        >
          Metadata
        </h2>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Metadata /> renders without crashing without links 1`] = `
<div>
  <div
    class="ui vertical ui overlay right very wide visible sidebar metadata-sidebar menu"
  >
    <div
      class="ui raised segments"
    >
      <div
        class="ui segment"
      >
        <h2
          class="ui header data-figure-block-header"
        >
          Metadata
        </h2>
      </div>
      <div
        class="ui segment"
      >
        <h3
          class="ui header data-figure-block-header"
        >
          Data Sources:
        </h3>
        <ul
          class="ui bulleted list widget"
          role="list"
        >
          <li
            class="horizontal"
            role="listitem"
          >
            <div>
              Test Data Sources
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Metadata /> renders without crashing without metadata 1`] = `
<div>
  <div
    class="ui vertical ui overlay right very wide visible sidebar metadata-sidebar menu"
  >
    <div
      class="ui raised segments"
    >
      <div
        class="ui segment"
      >
        <h2
          class="ui header data-figure-block-header"
        >
          Metadata
        </h2>
      </div>
    </div>
  </div>
</div>
`;

